pkg/kubelet/pod_workers.go

```diff
@@ -882,6 +907,8 @@ func (p *podWorkers) completeTerminatingRuntimePod(pod *v1.Pod) {
 	delete(p.podUpdates, pod.UID)
 	delete(p.lastUndeliveredWorkUpdate, pod.UID)
 	delete(p.terminatingStaticPodFullnames, kubecontainer.GetPodFullName(pod))
+
+	p.dispatchReservedStaticPodWork(kubecontainer.GetPodFullName(pod))
```

Comment:
Setting this up at the end is confusing.  I would simply prefer to spawn a regular pod worker for every pod (even if it's conflicting), and then have the pod manage loop check for conflicts (if it is blocked, simply queue a notification) and also have the termination of a pod wake any components it was flushing.  That avoids the need to have dispatchReservedStaticPodWork and having pod work queues get managed in two places (I would prefer that only UpdatePod queue work, and managePodLoop only requeue others).